### PR TITLE
Change push down logic in VIEW

### PR DESCRIPTION
**Pull request by 4ertus2**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Improve `enable_optimize_predicate_expression=1` logic for VIEW.

Detailed description / Documentation draft:
Push down columns for VIEW query in InterpreterSelectQuery instead of `StorageView.read()`. It allows to remove duplicate teble resolving there. It also apply all `analyze()` logic to VIEW subquery.

### CODE REVIEW

**Comment by 4ertus2** [comment]
Related #11127 #10481 #9424 (need tests after fix)

**Comment by 4ertus2** [comment]
It looks like it's not possible to run PredicateExpressionsOptimizer before join rewriter if thre're several joins in view.

File: src/Storages/StorageView.cpp [hunk 1]
```diff
@@ -87,60 +84,53 @@ Pipes StorageView::read(
...
-}
+    table_expression->subquery->setAlias(alias);
 
+    for (auto & child : table_expression->children)
+        if (child.get() == view_name.get())
+            child = subquery;
+    return subquery;
+}
 
-ASTPtr StorageView::getRuntimeViewQuery(ASTSelectQuery * outer_query, const Context & context, bool normalize)
```
Comment (4ertus2) [inline src/Storages/StorageView.cpp:120]: Remove additional table columns resolution here. Also remove PredicateExpressionsOptimizer call. There would be only one in SyntaxAnalyzer. Optimize VIEW subselect in InterpreterSelectQuery instead: full `analyze()` call, not only push down.

Negative side effect - ANALYZE for VIEW shows not optimized view query.

File: src/Interpreters/JoinToSubqueryTransformVisitor.cpp [hunk 2]
```diff
@@ -585,8 +585,9 @@ std::vector<TableNeededColumns> normalizeColumnNamesExtractNeeded(
     for (ASTIdentifier * ident : identifiers)
     {
         bool got_alias = aliases.count(ident->name);
+        bool allow_ambiguous = got_alias; /// allow ambiguous column overridden by an alias
 
-        if (auto table_pos = IdentifierSemantic::chooseTable(*ident, tables))
+        if (auto table_pos = IdentifierSemantic::chooseTableColumnMatch(*ident, tables, allow_ambiguous))
```
Comment (4ertus2) [inline src/Interpreters/JoinToSubqueryTransformVisitor.cpp:590]: The line that fixex the `multiple_joins_rewriter_version=2` trouble: `select t1.x, t2.x as x ...` is not an ambiguous case.

**Review by akuzm — APPROVED** [review]

**Pull request merged**